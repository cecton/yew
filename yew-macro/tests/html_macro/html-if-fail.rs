use yew::prelude::*;

fn compile_fail() {
    html! { if {html!()} };
    let value = 42;
    html! { if value { html!() } };
    html! { if {value} { html!() } };
    let boolean = true;
    html! { if boolean };
    html! { if boolean { () } };

    html! {
        <>
            <div/>
            if value {
            }
        </>
    };
}

fn main() {}
