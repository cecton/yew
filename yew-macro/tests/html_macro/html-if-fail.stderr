error: unexpected end of input, expected curly braces
 --> $DIR/html-if-fail.rs:4:5
  |
4 |     html! { if {} };
  |     ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected `if`
 --> $DIR/html-if-fail.rs:6:5
  |
6 |     html! { if true {} else };
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected curly braces
 --> $DIR/html-if-fail.rs:7:5
  |
7 |     html! { if true {} else if {} };
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected `if`
 --> $DIR/html-if-fail.rs:8:5
  |
8 |     html! { if true {} else if true {} else };
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> $DIR/html-if-fail.rs:5:16
  |
5 |     html! { if 42 {} };
  |                ^^ expected `bool`, found integer
